src/detect-modbus.c

```diff
@@ -191,13 +192,28 @@ static DetectModbus *DetectModbusAccessParse(DetectEngineCtx *de_ctx, const char
                 goto error;
 
             if (arg[0] == '>') {
-                modbus->address->min    = atoi((const char*) (arg+1));
+                if (StringParseUint16(&modbus->address->min, 10, 0,
```

Comment:
Is this three times the same code that can be factorized ?

Reply:
Except the last block. It uses `arg` instead of `arg + 1`.
How do you like the changes below? Please note that with these changes we assign `modbus->address->mode` before checking the value of `modbus->address->min`, I'm not sure if that would be a problem?
```
diff --git a/src/detect-modbus.c b/src/detect-modbus.c
index 9e412e759..f08d42d93 100644
--- a/src/detect-modbus.c
+++ b/src/detect-modbus.c
@@ -191,29 +191,21 @@ static DetectModbus *DetectModbusAccessParse(DetectEngineCtx *de_ctx, const char
             if (unlikely(modbus->address == NULL))
                 goto error;
 
+            int8_t idx;
             if (arg[0] == '>') {
-                if (StringParseUint16(&modbus->address->min, 10, 0,
-                                      (const char*) (arg + 1)) < 0) {
-                    SCLogError(SC_ERR_INVALID_VALUE, "Invalid value for min "
-                               "address: %s", (const char *)(arg + 1));
-                    goto error;
-                }
+                idx = 1;
                 modbus->address->mode   = DETECT_MODBUS_GT;
             } else if (arg[0] == '<') {
-                if (StringParseUint16(&modbus->address->min, 10, 0,
-                                      (const char*) (arg + 1)) < 0) {
-                    SCLogError(SC_ERR_INVALID_VALUE, "Invalid value for min "
-                               "address: %s", (const char*)(arg + 1));
-                    goto error;
-                }
+                idx = 1;
                 modbus->address->mode   = DETECT_MODBUS_LT;
             } else {
-                if (StringParseUint16(&modbus->address->min, 10, 0,
-                                      (const char*) arg) < 0) {
-                    SCLogError(SC_ERR_INVALID_VALUE, "Invalid value for min "
-                               "address: %s", arg);
-                    goto error;
-                }
+                idx = 0;
+            }
+            if (StringParseUint16(&modbus->address->min, 10, 0,
+                                  (const char*) (arg + idx)) < 0) {
+                SCLogError(SC_ERR_INVALID_VALUE, "Invalid value for min "
+                           "address: %s", (const char *)(arg + idx));
+                goto error;
             }
             SCLogDebug("and min/equal address %d", modbus->address->min);
```

Reply:
The proposed change looks better indeed :-)